config/mime-multipart.go

```diff
@@ -0,0 +1,151 @@
+package config
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime"
+	"mime/multipart"
+	"net/mail"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/coreos/coreos-cloudinit/pkg"
+)
+
+type MimeMultiPart struct {
+	Scripts []*Script
+	Config  *CloudConfig
+}
+
+func (m *MimeMultiPart) AddScript(script []byte) error {
+	s, err := NewScript(string(script))
+	if err == nil {
+		m.Scripts = append(m.Scripts, s)
+	}
+	return err
+}
+
+func (m *MimeMultiPart) AddCloudConfig(config []byte) error {
+	c, err := NewCloudConfig(string(config))
+	if err == nil {
+		if m.Config == nil {
+			m.Config = c
+		} else {
+			m.Config.Merge(c)
+		}
+	}
+	return err
+}
+
+func (m *MimeMultiPart) AddPlainText(pt []byte) error {
+	switch {
+	case IsScript(string(pt)):
+		return m.AddScript(pt)
+	case IsCloudConfig(string(pt)):
+		return m.AddCloudConfig(pt)
+	}
+	return nil
+}
+
+func (m *MimeMultiPart) fileExists(file string) bool {
+	_, err := os.Stat(file)
+	return err == nil
+}
+
+func (m *MimeMultiPart) readUrl(url string, once bool) ([]byte, error) {
+	var data []byte
+	var err error
+	workspace := "/var/lib/coreos-cloudinit" //TODO: read it from env
```

Comment:
I'd probably just back out the x-include-once-url. Its not needed because users can always just work around if its not there by making sure things are idempotent, which is more reliable anyways. 

If we added in this feature we'd also have to do it properly and pass the env down. Passing the env.Workspace down all the way from ParseUserData to here would add a decent amount of ugliness/complexity also without added anything needed.